Feature/mac gnu

natalie-perlin opened the pull request:
Added functionality for building on MacOS X (both M1/arm64 and Intel/x86_64)

Commit:
Initial changes for mac M1 arm64 architecture, GNU compilers gcc@11, openmpi-4.1.2

Commit:
added mac M1 functionality

Commit:
naming convention change to mac_m1

natalie-perlin committed:
Merge pull request #1 from natalie-perlin/feature/mac_m1_gnu

Feature/mac m1 gnu

Commit:
libs/build_madis.sh, build_mapl.sh - remove flags

Commit:
added stack_mac_m1.yaml

natalie-perlin committed:
Merge branch 'NOAA-EPIC:feature/mac_m1_gnu' into feature/mac_m1_gnu

natalie-perlin committed:
Merge pull request #2 from natalie-perlin/feature/mac_m1_gnu

Feature/mac m1 gnu - few minor cleanup updates for mac_m1_gnu

Commit:
Two separate configurations for Macs: x86_64 and mac_m1, config_<machine>.sh and stack_<machine>.yaml files

Commit:
Fix configure file for mac m86_64

Commit:
restored a configuration for macOS with mpich and clang

natalie-perlin committed:
Merge pull request #412 from NOAA-EPIC/feature/mac_m1_gnu

Feature/mac m1 gnu

natalie-perlin committed:
Update config_mac_m1_gnu.sh

natalie-perlin committed:
Update config_mac_m1_gnu.sh

kgerheiser commented on stack/stack_mac.yaml:
```diff
@@ -62,8 +62,8 @@ nccmp:
-  build: YES
-  version: 5.0.6
+  build: NO
+  version: 4.9.3
```
Why do you lower the version here? I've had success building with 5.0.6.

natalie-perlin replied:
Higher version could be left in the stack.yaml, as it is not being built on the MacOS. There is still an issue with the nco build on the MacOS

climbfuji replied:
We've built 5.0.1 and 5.0.6 on macOS with clang+gfortran as well as gcc+gfortran.

natalie-perlin replied:
Tested building with nco v.5.0.6 - worked OK! Placing it back to the stack.

kgerheiser commented on stack/stack_mac.yaml:
```diff
@@ -277,7 +277,7 @@ pybind11:
 ecbuild:
   build: YES
-  version: 3.6.1
+  version: 2021.08.0
```
Do you know what the difference is between the year version and the v3.x.x? I see they have a bunch of tags that kind of alternate between.

natalie-perlin replied:
I could not yet find any information about the differences. The tag 2021.08.0 seems to correspond to 3.6.3 (maybe identical?); a VERSION file in 2021.08.0 has 3.6.3, and also these tags seem to be similar in github, at least from the first glance.

natalie-perlin replied:
Tested the software build for fms 3.6.3 -successful. This could be reverted to 3.x.x format.

kgerheiser commented on stack/stack_mac.yaml:
```diff
@@ -322,24 +323,30 @@ yafyaml:
 miniconda3:
   build: NO
-  version: 4.7.12
+  version: 4.6.14
```
What's the reason for lowering the version?

kgerheiser replied:
What about the version of miniconda3 here? Why go from 4.7 to 4.6?

natalie-perlin replied:
It could be updated to 4.7.12 ( not building )

kgerheiser replied:
Nevermind, I this is fine. It's the version that `stack_noaa` uses, and it's not building.

kgerheiser commented on stack/stack_mac_gnu.yaml:
```diff
@@ -0,0 +1,359 @@
+gnu:
```
Can you say how this version differs specifically for GNU, compared to stack_mac?

natalie-perlin replied:
I did not use stack_mac.yaml, maybe it could be reserved for using it with clang+gfortran. Yet to be tested with this combination.

natalie-perlin replied:
Update of the compiler suite to 11.2.x . Note that there is a bug in 11.1.0 compiler that prevents building of the w3nco library, v. 4.2.1. in particular. This compiler bug that affects the w3nco software is reported in 
[https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100755](url)
[https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100283](url)
And it is resolved in the GNU 11.2.x versions.

kgerheiser commented on stack/stack_mac_m1_gnu.yaml:
```diff
@@ -0,0 +1,359 @@
+gnu:
```
Same for this

natalie-perlin replied:
This is only for GNU collection, when it is installed in advance of the build. MPI could be installed during the hpc-stack build.

natalie-perlin committed:
Update stack_mac.yaml

natalie-perlin committed:
Update stack_mac_m1_gnu.yaml

nco build "YES" v.5.0.6

natalie-perlin committed:
Update stack_mac_gnu.yaml

nco build "YES" , v.5.0.6

natalie-perlin committed:
Update stack_mac_m1_gnu.yaml

natalie-perlin committed:
Update stack_mac_gnu.yaml

natalie-perlin committed:
Update stack_mac.yaml

natalie-perlin committed:
Update stack_mac_m1_gnu.yaml

natalie-perlin committed:
Update stack_mac_gnu.yaml

natalie-perlin committed:
Update stack_mac.yaml

fms version 2021.03 - to make it more consistent with the version of the library used for the UFS

kgerheiser committed:
Merge branch 'develop' into feature/mac_gnu

kgerheiser commented:
I resolved the conflict in your branch

natalie-perlin committed:
Merge branch 'NOAA-EMC:feature/mac_gnu' into feature/mac_gnu

natalie-perlin committed:
Update stack_mac_m1_gnu.yaml

natalie-perlin committed:
Update stack_mac.yaml

natalie-perlin committed:
Update stack_mac_gnu.yaml

natalie-perlin reviewed (commented):
All issues have been addressed.

It still says the merging is blocked. I will try updating a branch on NOAA-EPIC/hpc-stack:mac_gnu branch, then open pull requests from NOAA-EPIC to NOAA-EMC mac_gnu branch, and from mac_gnu to develop on NOAA-EMC side.
![MergingBlocked](https://user-images.githubusercontent.com/68030316/160698896-47ec46b0-14be-4db9-9e55-6c4e8d0cb56b.png)

natalie-perlin commented:
All questions addressed, but some technical issues preventing merging ("merging blocked") within this pull request

kgerheiser commented:
I just had to re-approve the PR. Anyway #420 is open, so I'll use that.

natalie-perlin committed:
Merge pull request #420 from NOAA-EPIC/feature/mac_gnu

Feature/mac gnu, Intel/x86_64 and M1/arm64, with software+libraries updated to higher versions

natalie-perlin commented:
Reopened the PR, after updating of the feature/mac_gnu, it could be merged with the develop if all goes well:
 `NOAA-EMC:feature/mac_gnu` into `NOAA-EMC:develop`
